org.eclipse.dartboard/src/org/eclipse/dartboard/preference/DartSDKLocationFieldEditor.java

```diff
@@ -17,15 +29,85 @@ public DartSDKLocationFieldEditor(String preferencesKey, String label, Composite
 	@Override
 	protected boolean doCheckState() {
 		String location = getTextControl().getText();
-		File file = new File(location);
-		boolean isValid = file.exists();
+		boolean isValid = isValidDartSDK(location);
 		if (!isValid) {
 			setErrorMessage(Messages.Preference_SDKNotFound_Message);
 			showErrorMessage();
 		}
 		return isValid;
 	}
 
+	/**
+	 * Checks if a given path is the root directory of a Dart SDK installation.
+	 * 
+	 * Returns false if the path does not exist or the given location can not be
+	 * converted to a {@link Path}.
+	 * 
+	 * Similarly if the Path is not a directory, false is returned.
+	 * 
+	 * If the location is a symbolic link but it can not be resolved, false is
+	 * returned.
+	 * 
+	 * If the process to test the version string returned by the Dart executable can
+	 * not be executed, false is returned.
+	 * 
+	 * Finally, if the returned version string does not start with "Dart VM
+	 * version", false is returned.
+	 * 
+	 * @param location - A {@link String} that should be checked to be a Dart SDK
+	 *                 root directory.
+	 * @return <code>false</code> if the location is not a Dart SDK root directory,
+	 *         <code>true</code> otherwise.
+	 */
+	@SuppressWarnings("nls")
+	private boolean isValidDartSDK(String location) {
+		boolean isWindows = Platform.OS_WIN32.equals(Platform.getOS());
+		Path path = null;
+		// On Windows if a certain wrong combination of characters are entered a
+		// InvalidPathException is thrown. In that case we can assume that the location
+		// entered is not a valid Dart SDK directory either.
+		try {
+			path = Paths.get(location).resolve("bin" + File.separator + (isWindows ? "dart.exe" : "dart"));
+		} catch (InvalidPathException e) {
+			return false;
+		}
+		// If the entered file doesn't exist, there is no need to run it
+		// Similarly if the file is a directory it can't be the dart executable
+		if (!Files.exists(path) || Files.isDirectory(path)) {
+			return false;
+		}
+		// Follow symbolic links
+		try {
+			path = path.toRealPath();
+		} catch (IOException e1) {
+			LOG.error("Couldn't follow symlink", e1);
+			return false;
+		}
+
+
+		String executablePath = path.toAbsolutePath().toString();
+
+		String[] commands;
+		if (isWindows) {
+			commands = new String[] { "cmd", executablePath + ".exe --version" };
```

Comment:
Change this line to 			commands = new String[] { "cmd", "/c", executablePath, "--version" };